src/components/Tx/TxHistory/TxEntry.jsx

```diff
@@ -0,0 +1,349 @@
+          </span>
+        </span>
+      )
+    }
+
+    return (
+      <StyledTx key={tx.hash || tx.nonce}>
+        <div className="right">
+          {network && <div className="network">{network}</div>}
+          <div className="date">{date}</div>
+        </div>
+        <div className="description">{description}</div>
+        {tx.contractAddress && (
+          <div>
+            <div className="label">{i18n.t('mist.txHistory.newContract')}</div>
+            <div className="identicon">
+              <Identicon address={tx.contractAddress} size="small" />
+            </div>
+            <span className="value">{tx.contractAddress}</span>
+          </div>
+        )}
+        <div>
+          <div className="label">{i18n.t('mist.txHistory.from')}</div>
+          <div className="identicon">
+            <Identicon address={tx.from} size="small" />
+          </div>
+          <span className="value">{tx.from}</span>
+        </div>
+        {isTokenTransfer && (
+          <div>
+            <div className="label">{i18n.t('mist.txHistory.to')}</div>
+            <div className="identicon">
+              <Identicon address={tokensTo} size="small" />
+            </div>
+            <span className="value">{tokensTo}</span>
+          </div>
+        )}
+        {!isTokenTransfer && (
+          <div>
+            {tx.to && (
+              <div>
+                <div className="label">
+                  {i18n.t('mist.txHistory.to')}
+                  {tx.toIsContract && ` ${i18n.t('mist.txHistory.contract')}`}
+                </div>
+                <div className="identicon">
+                  <Identicon address={tx.to} size="small" />
+                </div>
+                <span className="value">{tx.to}</span>
+              </div>
+            )}
+          </div>
+        )}
+        <div>
+          <div className="label">{i18n.t('mist.txHistory.status')}</div>
+          <strong className="value">{status}</strong>
+        </div>
+        {this.renderDetails()}
+      </StyledTx>
+    )
+  }
+}
+
+const StyledTx = styled.div`
+  background: #f9f9f9;
+  margin-bottom: 1em;
+  border-bottom: solid 2px #dcdada;
+  padding: 1em;
+  color: #111;
+  word-wrap: break-word;
+
+  .description {
+    font-size: 125%;
+    margin-bottom: 0.5em;
+  }
+
+  .label {
+    display: inline-block;
+    background: #ededed;
+    color: #444;
+    padding: 0.35em;
+    border-radius: 5px;
+    font-size: 80%;
+    margin-right: 5px;
+  }
+
+  .right {
+    float: right;
+    text-align: right;
+  }
+
+  .date {
+    font-size: 80%;
+    margin-bottom: 0;
+  }
+
+  .network {
+    color: red;
+    font-size: 13px;
+    font-weight: value;
+    margin-bottom: 0;
+  }
+
+  .identicon {
+    display: inline-block;
+    vertical-align: middle;
+    margin: 0 7px 0 2px;
+  }
+
+  div {
+    margin-bottom: 10px;
+  }
+
+  a {
+    text-decoration: none;
+  }
+
+  .moreDetails {
```

Comment:
I'm still unsure of best practices in styled-components. Couple open-ended questions:
1. Should a styled component contain many children styles (like you've done here), or should each of those child elements be styled as separate JS objects? (I've been doing the latter)
2) How much style configurability should we permit? We're close to "none" on the spectrum and maybe that's okay.

Reply:
1. This was just to ease the transition, since the component was already coded. I think that all new code being written should have child elements styled as separate JS objects. I can convert this component if it's a good use of time, but thought there might be other priorities.

2. I like the idea of including style configurability, but maybe it can start being added on an "as-needed" basis.